addons/account/models/account_account.py

```diff
@@ -1229,6 +1230,292 @@ def _action_merge(self, account_to_merge_into, code_by_company):
+                """,
+                table=SQL.identifier(self.env[model]._table),
+                column=SQL.identifier(field_to_update.name),
+                new_account_id_by_company_id_json=new_account_id_by_company_id_json,
+                company_id_column=SQL.identifier(company_id_field),
+                value_to_update=f'account.account,{self.id}',
+                company_ids_to_update=tuple(companies_to_update.ids),
+            )
+            operations[model].append((field_to_update, records_to_update, query))
+
+        # 3.3: Add Many2OneReference fields that reference account.account
+        many2one_reference_fields = self.env['ir.model.fields'].search([
+            ('ttype', '=', 'many2one_reference'),
+            ('store', '=', True),
+            '!',
+            ('model', '=', 'studio.approval.request'),  # A weird Many2oneReference which doesn't have its model field on the model.
+            ('name', '=', 'res_id'),
+        ])
+        for field_to_update in many2one_reference_fields:
+            model = field_to_update.model
+            model_field = self.env[model]._fields[field_to_update.name]._related_model_field
+            if not self.env[model]._auto or not self.env[model]._fields[model_field].store:
+                continue
+            if model == 'res.company':
+                company_id_field = 'id'
+            elif 'company_id' in self.env[model]:
+                company_id_field = 'company_id'
+            else:
+                continue
+            if not self.env[model]._fields[company_id_field].store:
+                continue
+            records_to_update = self.env[model].search([(field_to_update.name, '=', self.id), (model_field, '=', 'account.account')])
+            query = SQL(
+                """
+                 UPDATE %(table)s
+                    SET %(column)s = (%(new_account_id_by_company_id_json)s::jsonb->>%(company_id_column)s::text)::int
+                  WHERE %(column)s = %(account_id)s
+                    AND %(model_column)s = 'account.account'
+                    AND %(company_id_column)s IN %(company_ids_to_update)s
+                """,
+                table=SQL.identifier(self.env[model]._table),
+                column=SQL.identifier(field_to_update.name),
+                new_account_id_by_company_id_json=new_account_id_by_company_id_json,
+                company_id_column=SQL.identifier(company_id_field),
+                account_id=self.id,
+                model_column=SQL.identifier(model_field),
+                company_ids_to_update=tuple(companies_to_update.ids),
+            )
+            operations[model].append((field_to_update, records_to_update, query))
+
+        # 3.4: Execute UPDATEs
+        for model, model_operations in operations.items():
+            for field_to_update, records_to_update, query in model_operations:
+                records_to_update.invalidate_recordset()
```

Comment:
Maybe this is what you wanted?
```suggestion
                records_to_update.invalidate_recordset([field_to_update])
```

Reply:
Yes! :+1: